tagController.js

```diff
@@ -20,47 +20,57 @@ function tagAddRow(){
-    newTagRowCell0.appendChild(cell0Element);
+    let cell0_RemoveLabel = document.createElement("label");
+    cell0_RemoveLabel.id = newTagRow.id + "_remove_label";
+    newTagRowCell0.appendChild(cell0_RemoveLabel);
 
     //creates dropdown for TAG row
-    let cell1Element = document.createElement("select");
-    tagList("tagRow_" + newTagRowId + "_select", cell1Element);
-    cell1Element.onchange = function(){tagRowSelectUpdate(newTagRowId)};
-    cell1Element.onfocus = function(){tagListOnFocus(this)};
-    newTagRowCell1.appendChild(cell1Element);
+    let cell1_TagSelect = document.createElement("select");
+    tagList("tagRow_" + newTagRowId + "_select", cell1_TagSelect);
+    cell1_TagSelect.onchange = function(){tagRowSelectUpdate(newTagRowId)};
+    cell1_TagSelect.onfocus = function(){tagListOnFocus(this)};
+    newTagRowCell1.appendChild(cell1_TagSelect);
 
     //creates the 'rank' settings
-    let cell2Element = document.createElement("label");
-    cell2Element.id = "tagRow_" + newTagRowId + "_rank";
-    cell2Element.value = 0;
-    cell2Element.innerText = '0/[max]';
-    newTagRowCell2.appendChild(cell2Element);
+    let cell2_TagRank = document.createElement("label");
+    cell2_TagRank.id = "tagRow_" + newTagRowId + "_rank";
+    cell2_TagRank.value = 0;
+    cell2_TagRank.innerText = '0/[max]';
+    newTagRowCell2.appendChild(cell2_TagRank);
 
     //creates the 'active' checkbox
-    let cell3Element = document.createElement("input");
-    cell3Element.id = "tagRow_" + newTagRowId + "_checkBox";
-    cell3Element.type = "checkbox";
-    cell3Element.onclick = function(){tagRowChangeActive(cell3Element, cell4Element)};
-    newTagRowCell3.appendChild(cell3Element);
+    let cell3_TagActiveCheck = document.createElement("input");
+    cell3_TagActiveCheck.id = "tagRow_" + newTagRowId + "_checkBox";
+    cell3_TagActiveCheck.type = "checkbox";
+    cell3_TagActiveCheck.onclick = function(){tagRowChangeActive(newTagRowId)};
+    newTagRowCell3.appendChild(cell3_TagActiveCheck);
 
     //creates the 'cost' value box
-    let cell4Element = document.createElement("label");
-    cell4Element.id =  "tagRow_" + newTagRowId + "_costBox";
-    cell4Element.innerText = '0';
-    newTagRowCell4.appendChild(cell4Element);
+    let cell4_TagCost = document.createElement("label");
+    cell4_TagCost.id =  "tagRow_" + newTagRowId + "_costBox";
+    cell4_TagCost.innerText = '0';
+    newTagRowCell4.appendChild(cell4_TagCost);
 
     //Description section
-    let cell5Element = document.createElement("div");
-    cell5Element.id = "tagRow_" + newTagRowId + "_desc";
-    cell5Element.innerText = tagDataLocalList().tags[cell1Element.value].desc;
-    newTagRowCell5.appendChild(cell5Element);
+    let cell5_TagDesc = document.createElement("div");
+    cell5_TagDesc.id = "tagRow_" + newTagRowId + "_desc";
+    cell5_TagDesc.innerText = tagDataLocalList().tags[cell1_TagSelect.value].desc;
+    newTagRowCell5.appendChild(cell5_TagDesc);
```

Comment:
This really wants to be a data structure, not a bunch of vars

Reply:
shhhh... _sooooon_